src/rust/ide/src/view/node_editor.rs

```diff
@@ -657,33 +700,32 @@ impl GraphEditorIntegratedWithControllerModel {
 // === NodeEditor ===
 // ==================
 
-/// Node Editor Panel integrated with Graph Controller.
+/// Node Editor Panel integrated with Synchronized Executed Graph Model.
 #[derive(Clone,CloneRef,Debug)]
 pub struct NodeEditor {
     logger         : Logger,
     display_object : display::object::Instance,
     #[allow(missing_docs)]
-    pub graph     : Rc<GraphEditorIntegratedWithController>,
-    controller    : controller::ExecutedGraph,
+    pub graph     : Rc<GraphEditorIntegratedWithModel>,
     visualization : controller::Visualization
 }
 
 impl NodeEditor {
     /// Create Node Editor Panel.
     pub async fn new
-    ( logger        : impl AnyLogger
-    , app           : &Application
-    , controller    : controller::ExecutedGraph
-    , project       : controller::Project
-    , visualization : controller::Visualization) -> FallibleResult<Self> {
+    (logger        : impl AnyLogger
+     , app           : &Application
+     , model         : model::ExecutedGraph
+     , project       : controller::Project
+     , visualization : controller::Visualization) -> FallibleResult<Self> {
         let logger         = Logger::sub(logger,"NodeEditor");
         let display_object = display::object::Instance::new(&logger);
-        let graph          = GraphEditorIntegratedWithController::new(logger.clone_ref(),app,
-            controller.clone_ref(),project);
+        let graph          = GraphEditorIntegratedWithModel::new(logger.clone_ref(), app, model,
```

Comment:
Please do not create API which requeires callside to clone logger. Instead consume it as AnyLogger iface

Reply:
I'm not convinced — the `GraphEditorIntegratedWithModel` takes a logger to store it, not to `sub` it. 
In such case i'd say that accepting the Logger rather than AnyLogger is the right thing to do.
(as we care e.g. about the state of that particular logger and we don't want to break it by converting back and forth)